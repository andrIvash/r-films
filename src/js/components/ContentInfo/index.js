// @flow
import React from 'react';
import FilmItem from '../FilmItem';
import { Film } from '../../flow-types.js';

type Props = {
 onFilmSelect: (id: number) => void,
 films: Array<Film>
}

function filmItems(props: Props) {
  if (props.films && props.films.length) {
    return props.films.map((item, ndx) => (
      <li
        className='film-list__item'
        key={item.id}
        >
        <FilmItem
          film={item}
          id={item.id}
          onFilmSelect={props.onFilmSelect}
        />
      </li>
    ));
  }
  return <h2 className='no-data'>No films found</h2>;
}

const ContentInfo = (props: Props) => (
  <div className='content-info'>
    <ul className='film-list'>

      {filmItems(props)}

      {props.films.length && props.films.length ? (
        props.films.map(film => <FilmItem {...film} onFilmSelect={props.onFilmSelect} />)
      ) : (
        <h2 className='no-data'>No films found</h2>
      )}
    </ul>
  </div>
);

export default ContentInfo;
