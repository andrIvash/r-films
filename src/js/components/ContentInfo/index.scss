.content-info {
  width: 100%;
}

.film-list {
  margin: rem(20px) 0;
  list-style: none;
  padding: 0;
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
  align-items: center;

  @include tablet() {
    justify-content: center;
  }
}

.no-data {
  color: $color-background;
  font-family: $sign-font;
  font-weight: 700;
  font-size: rem(40px);
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-top: rem(-30px);
  @include centered();
  top: rem(-80px);
}

.film-list__item {
  width: rem(225px);
  min-height: rem(380px);
  border: 1px solid black;
  margin: rem(15px);
}