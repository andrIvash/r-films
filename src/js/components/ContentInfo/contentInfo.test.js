import React from 'react';
import ContentInfo from './index';

describe('ContentInfo', () => {
  it('should be defined', () => {
    expect(ContentInfo).toBeDefined();
  });
});
