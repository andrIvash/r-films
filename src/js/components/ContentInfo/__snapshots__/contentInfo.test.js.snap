// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentInfo should render correctly 1`] = `
<div
  class="content-info"
>
  <ul
    class="film-list"
  >
    <filmitem
      film="[object Object]"
      id="1"
    />
    <filmitem
      film="[object Object]"
      id="2"
    />
  </ul>
</div>
`;
